x-pack/plugin/enrich/src/test/java/org/elasticsearch/xpack/enrich/EnrichPolicyRunnerTests.java

```diff
@@ -255,6 +255,93 @@ public void testRunnerGeoMatchType() throws Exception {
         ensureEnrichIndexIsReadOnly(createdEnrichIndex);
     }
 
+    public void testRunnerMatchTypeWithIpRange() throws Exception {
+        final String sourceIndexName = "source-index";
+        createIndex(sourceIndexName, Settings.EMPTY, "_doc", "subnet", "type=ip_range");
+        IndexResponse indexRequest = client().index(
+            new IndexRequest().index(sourceIndexName)
+                .id("id")
+                .source("{" + "\"subnet\":" + "\"10.0.0.0/8\"," + "\"department\":\"research\"" + "}", XContentType.JSON)
```

Comment:
Can we test the scenario where `subnet` and `department` are fields underneath an object field (like `organization.subnet` and `organization.department`)?

Reply:
Added additional tests, did reveal an issue with the code. Good one.